pkg/config/generate.go

```diff
@@ -59,18 +59,46 @@ func selectContainerImagesOrigin(cs *acsapi.ContainerService, c *Config) {
 	}
 }
 
+func selectContainerImagesOSA(cs *acsapi.ContainerService, c *Config) {
+	switch cs.Properties.OrchestratorProfile.OpenShiftConfig.OpenShiftVersion {
+	//TODO: confirm minor version after release
+	case "3.10", "3.10.15":
```

Comment:
this seems like it's going to a) get pretty repetitive and b) need a lot of updates.  What can we do here to make it easier to maintain?  Suggestions:

* semver comparison
* convention based on version for majority of images (not sure how possible this is)
* slice of images that take a coordinated version number (all the 3.10.15-1) and a helper that sets all of them that takes just a version

combination of the above, something completely different?

Reply:
Not a blocker since we have a data set of 1 so far :smile: 

Probably want a panic or error if we're not setting images though so we don't move along

Reply:
:+1:

Reply:
Question is how granular this will be. We would need to do some defaulting. Let's say a customer chose `3.10.` and the newest one is `3.10.15`, so we default to this. And we just got new image set (`3.10.20`). If we are running some arbitrary helm code (change secret permissions or deployment parameters) which does not involve image change we will need to make sure that we are running with `3.10.15` metadata for that run. Because I assume we want to control when we do minor patches upgrades so we would not have unplanned updated. 

Overall this will comes to decide if we want to be able to do deployment config/helm changes and upgrade changes as a separate action.

Reply:
> openShiftVersion is a user-facing option and I think we shouldn't allow users to choose patch versions (ie. we should always update them to latest).

Seems like we need to support n versions even if the user cannot pick it so that we can perform upgrades correctly.  Limiting what the user can choose seems separate than the image selection.

Reply:
agreed. my biggest current `unknown` is how we make sure we dont do unintentional upgrades (even minor) when running sync container for lets say configuration change and not upgrade. This means sync will have to be aware of *current running version* and what we are asking it to do. Or im overthinking this one?

Reply:
sync is reading both the manifest and existing config.yaml.  The config.yaml has the image configuration already there.  Sync isn't running through the Generate portions.

Reply:
> Limiting what the user can choose seems separate than the image selection.

Agreed, my comment was specifically about handling openshiftVersion which should never contain a patch version.

Reply:
changed :)